### PR TITLE
Feat/add template

### PR DESCRIPTION
**Pull request by hwan513**
### Changes:
Describe the Pull Request here. Add any references and info to help reviewers understand your changes.
- Just creating pr first I'll add the details in the morning and also add the related task

### Tests Applied:
What testing has been applied to validate changes are accurate and correct
- No testing applied

### Issue ticket number(s):
Enter the issue numbers resolved by this pull request
- #12 

### Checklist
- [x] Documented
- [x] Linting tests successful
- [x] merge updated prod branch into development branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
 
 The tool works locally; font files are not uploaded to any server.
 
+## Development
+
+This project uses React for Front-end, Vanilla CSS for styling, and Vite for server tooling.
```
Comment (jaskfla) [inline README.md]:
```suggestion
This project uses React for front-end, vanilla CSS for styling, and Vite for server tooling.
```

File: README.md [hunk 2]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+
+[Node.js](https://nodejs.org/en) 16+ is recommended for development as anything less has yet to be tested.
```
Comment (jaskfla) [inline README.md]:
```suggestion
[Node.js](https://nodejs.org) 16+ is recommended for development; Typefaceoff is not tested with older versions.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+The package manager of choice is pnpm instead of npm or yarn. Installation instructions [here](https://pnpm.io/installation).
```
Comment (jaskfla) [inline README.md]:
```suggestion
The package manager of choice is pnpm—installation instructions [here](https://pnpm.io/installation).
```

File: README.md [hunk 4]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+The following commands should enable you to set up a development server running of the website.
+
+```sh
+# clone your fork of the git repository and open it
```
Comment (jaskfla) [inline README.md]:
```suggestion
# Clone your fork of the Git repository and open it
```

File: README.md [hunk 5]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+cd typefaceoff
+# install dependencies with pnpm (see docs if not installed https://pnpm.io/installation)
```
Comment (jaskfla) [inline README.md]:
```suggestion
# Install dependencies with pnpm
# (Consult https://pnpm.io/installation if you don’t have pnpm installed)
```

File: README.md [hunk 6]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+pnpm install
+# set up pre-commit hooks for formatting
```
Comment (jaskfla) [inline README.md]:
```suggestion
# Set up pre-commit hooks for code formatting
```

File: README.md [hunk 7]
```diff
@@ -4,6 +4,38 @@ Typefaceoff is a a drag-and-drop tool for proofing and comparing typefaces in a
...
+pnpm run husky
+# start the dev server
```
Comment (jaskfla) [inline README.md]:
```suggestion
# Start the development server
```

File: index.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/src/main.tsx"></script>
```
Comment (jaskfla) [inline index.html:11]: Any reason this isn’t in `<head>` with `defer`? I’m not familiar with TypeScript, so don’t know if my approach actually works.

File: src/App.css [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  max-width: 1280px;
+  margin: 0 auto;
+  padding: 2rem;
+  text-align: center;
+}
+
+.logo {
+  height: 6em;
+  padding: 1.5em;
+  will-change: filter;
```
Comment (jaskfla) [inline src/App.css:11]: Is this `will-change` automatically generated? If not, why do we need it?

In general, using `will-change` should be a last resort. In *almost* every scenario we should let the browser decide how to optimise its resources.

File: src/index.css [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
+  line-height: 1.5;
+  font-weight: 400;
+
+  color-scheme: light dark;
+  color: rgba(255, 255, 255, 0.87);
+  background-color: #242424;
+
+  font-synthesis: none;
+  text-rendering: optimizeLegibility;
```
Comment (jaskfla) [inline src/index.css]: Given this tool is for proofing fonts, I’d argue being true to the letterforms is more important than alignment.

```suggestion
  text-rendering: geometricPrecision;
```

File: src/index.css [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+body {
+  margin: 0;
+  display: flex;
+  place-items: center;
+  min-width: 320px;
+  min-height: 100vh;
+}
+
+h1 {
+  font-size: 3.2em;
```
Comment (jaskfla) [inline src/index.css]:
```suggestion
  font-size: 3.2rem;
```

File: src/index.css [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  line-height: 1.1;
+}
+
+button {
+  border-radius: 8px;
+  border: 1px solid transparent;
+  padding: 0.6em 1.2em;
+  font-size: 1em;
```
Comment (jaskfla) [inline src/index.css]:
```suggestion
  font-size: 1rem;
```

File: .prettierrc [hunk 13]
```diff
@@ -0,0 +1,6 @@
+{
+  "printWidth": 80,
```
Comment (jaskfla) [inline .prettierrc]: Even relatively small screens can handle a little wider than 80.

```suggestion
  "printWidth": 100,
```

**Review by jaskfla — COMMENTED** [review]
Mostly stylistic nitpicks.

I’m trusting that the pre-commit hook and actual server stuff works. Probably best if someone else (with experience) also reviews this PR.

File: index.html (hunk 8 above)
Comment (hwan513) [reply index.html:11]: This is how it was generated by the template. I did try moving it into the `<head>` and it still worked fine.

File: src/App.css (hunk 9 above)
Comment (hwan513) [reply src/App.css:11]: Part of the template.

**Review by cste303 — APPROVED** [review]
Managed to get it working easily on my side using the steps provided. All looks good.

**Pull request merged**